query_temp.py

```diff
@@ -1,186 +1,113 @@
-                                 Obs.time < end_time))
-                    .filter(func.extract("month", Obs.time) == month)
-                    .filter(Variable.name == '2')
-                    .filter(and_(Variable.standard_name == 'air_temperature',
-                                 Variable.cell_method == 'time: minimum'))
-                    .filter(Obs.datum != 0.0) # bad obs are sometimes 0.0
-                    .group_by(History.lat,
-                              History.lon,
-                              History.station_id)
-             )
+    # arrange data depending on upper or lower percentile
+    order_dict = {'lower': Obs.datum.asc(),
+                  'upper': Obs.datum.desc()}
 
-    return query
+    if method not in order_dict.keys():
+        raise ValueError('Please enter a valid method type. Must be \'lower\' or \'upper\', ')
 
+    order = order_dict[method]
 
-def query_design_temp_dry(start_time, end_time, session, quantile=0.025):
-    """A query to get the percentile of maximum daily
-    air_temperatures (it is assumed that dry bulb temperatures are
-    identical to regular air temperatures) at a station in a
-    given time frame. Daily maximum air temperatures are used.
-    ---------------------------------------------------------
-    Args:
-        session (sqlalchemy Session): session constructed using
-            connection string and engine
-        percentile (float, optional): desire percentile in fraction.
-            Default value is 0.0025 or the 2.5th percentile
-        month (int, optional): desired month in which to calculate
-            the desired percentile, default is 7, or July.
-    Returns:
-        query (sqlalchemy query): sqlalchemy query object
-            for design value
-    """
-
+    # percentile calculation
     percentile = (func.percentile_cont(quantile)
-                 .within_group(Obs.datum.desc())
-                 .label("air_temperature"))
-    days = days_in_month(start_time, end_time)
-    completeness = (count(Obs)/days).label("completeness")
-
-    month = start_time.month
-    # construct query table
-    query = (
-             session.query(percentile,
-                           min_time(Obs),
-                           max_time(Obs),
-                           History.lat,
-                           History.lon,
-                           History.station_id,
-                           completeness)
+                      .within_group(order)
+                      .label("air_temperature"))
+
+    # This query has eligible stations with fully complete
+    # record over month of interest
+    cte_full_cvg = (session.query(
+                            completeness,
+                            History.station_id.label('station_id'),
+                            func.extract('year', Obs.time).label('year')
+                            )
+                           .join(Variable, Obs.vars_id == Variable.id)
```

Comment:
Especially with a query of this complexity and indirectness (by that, I refer to items like `completeness`, which mention a table and label a column, but via a variable), it is important to be explicit what tables are being selected from. Therefore, specify that you are selecting from `Obs` too.